src/main/groovy/org/labkey/gradle/plugin/JavaModule.groovy

```diff
@@ -185,43 +186,28 @@ class JavaModule implements Plugin<Project>
                     task.doFirst({
                         // first clean out the directory, if it existed from a previous build
                         if (destinationDir.exists()) // I think Windows doesn't deal well with delete without this check first
-                            destination.deleteDir();
+                            destination.deleteDir()
                     })
             }
 
             TaskUtils.configureTaskIfPresent(project, 'module', {
                 dependsOn(project.tasks.copyExternalLibs)
-                if (project.tasks.findByName("jar") != null)
-                {
-                    dependsOn(project.tasks.jar)
-                }
-                if (project.tasks.findByName('apiJar') != null)
-                {
-                    dependsOn(project.tasks.apiJar)
-                }
-                if (project.tasks.findByName('jspJar') != null)
-                {
-                    dependsOn(project.tasks.jspJar)
-                }
+                JAR_TASK_NAMES.forEach(taskName -> {
+                    TaskUtils.addOptionalTaskDependency(project, it, taskName)
+                })
+                if (LabKeyExtension.isDevMode(project))
+                    TaskUtils.addOptionalTaskDependency(project, it, "copyToModulesApi")
             } )
 
             project.tasks.register(
                     "checkModuleJarVersions", CheckForVersionConflicts)
                     { CheckForVersionConflicts task ->
 
                         FileCollection allJars = externalFiles
-                        if (project.tasks.findByName("jar") != null)
-                        {
-                            allJars = allJars + project.tasks.jar.outputs.files
-                        }
-                        if (project.tasks.findByName('apiJar') != null)
-                        {
-                            allJars = allJars + project.tasks.apiJar.outputs.files
-                        }
-                        if (project.tasks.findByName('jspJar') != null)
-                        {
-                            allJars = allJars + project.tasks.jspJar.outputs.files
-                        }
+                        for (String taskName : JAR_TASK_NAMES)
+                            TaskUtils.doIfTaskPresent(project, taskName, (jarTask) -> {
+                                allJars = allJars + jarTask.get().outputs.files
+                            })
```

Comment:
I didn't realize groovy doesn't share Java's restriction on accessing non-final variables within lambdas. Handy.

Reply:
Yeah. Still wish I hadn't chosen Groovy as the language here, though.